packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -461,11 +464,41 @@ export default function useAutocomplete(props) {
     },
   );
 
+  const checkHighlightedOptionExists = () => {
+    if (
+      highlightedIndexRef.current !== -1 &&
+      previousProps.filteredOptions &&
+      previousProps.filteredOptions.length !== filteredOptions.length &&
+      (multiple
+        ? previousProps.value.every((val, i) => getOptionLabel(value[i]) === getOptionLabel(val))
+        : getOptionLabel(previousProps.value ?? '') === getOptionLabel(value ?? ''))
+    ) {
+      const previousHighlightedOption = previousProps.filteredOptions[highlightedIndexRef.current];
+
+      if (previousHighlightedOption) {
+        const previousHighlightedOptionExists = filteredOptions.some((option) => {
+          return getOptionLabel(option) === getOptionLabel(previousHighlightedOption);
```

Comment:
But in the case when options are objects, it would always reset, with the strict equality always returning `false` because they point to different instances.

Reply:
> it would always reset

Why? I would expect developers to fetch new options, add them to the prop, and keep the same reference to the existing options. If they re-fetch all the options, maybe it's correct to reset the scroll?

Reply:
So you're saying that, consider in the test, [this](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Autocomplete/Autocomplete.test.js#L1681) is incorrect because developers re-fetch all the options (including a new one), so we should expect to reset? But shouldn't React developers treat objects and arrays as immutable when storing them in state (options and props)? But shouldn't React developers treat objects/arrays as immutable when storing in state (`options` prop)? Then it would be a new reference?

Reply:
Yes, in this test, these look like entirely new options. I think that a scroll reset can make sense.

#26492 makes the case that we should support duplicated labels.

Reply:
Actually, we could use `isOptionEqualToValue` to determine if two options are the same instead of relying on labels. This should also fix #36114